scripts/build_release_upload.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+. ${HOME}/build_env
+set -e
+set -x
+
+# persistent storage of repo manager scratch space is on EFS
+if [ ! -z "${AWS_EFS_MOUNT}" ]; then
+    if mount|grep -q /data; then
+	echo /data already mounted
+    else
+	sudo mkdir -p /data
+	sudo mount -t nfs4 -o nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=600,retrans=2,noresvport "${AWS_EFS_MOUNT}":/ /data
+	# make environment for release_deb.sh
+	sudo mkdir -p /data/_aptly
+	sudo chown -R ${USER} /data/_aptly
+	export APTLY_DIR=/data/_aptly
+    fi
+fi
+
+cd ${GOPATH}/src/github.com/algorand/go-algorand
+
+git push origin --tags ${BRANCH}
+
+cd ${PKG_ROOT}
+
+if [ ! -z "${S3_PREFIX}" ]; then
+    aws s3 sync --quiet --exclude dev\* --exclude master\* --exclude nightly\* --exclude stable\* --acl public-read ./ ${S3_PREFIX}/${CHANNEL}/${RSTAMP}_${FULLVERSION}/
+fi
+
+# copy .rpm file to intermediate yum repo scratch space, actual publish manually later
+if [ ! -d /data/yumrepo ]; then
+    sudo mkdir -p /data/yumrepo
+    sudo chown ${USER} /data/yumrepo
+fi
+cp -p -n *.rpm *.rpm.sig /data/yumrepo
```

Comment:
These need to be pushed to S3.  Promoting should be possible from any machine, which means all assets should already be in S3 -- promoting will mean moving them to the appropriate location.

Reply:
s3 upload happened a couple lines up with `aws s3 sync`. That maintains this build's output in a staging/archival way.
/data/ is an EFS (AWS managed NFS) volume for persistent scratch space for `aptly` and the rpm `createrepo` tool; once they lay out files as an apt or yum repo those get published to the serving s3 bucket.
`aptly` can probably run anywhere since it is a pure Go tool. `createrepo` is probably best from Centos as either a VM or docker container.